src/targets/gpu/include/migraphx/gpu/event.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef MIGRAPHX_GUARD_MIGRAPHLIB_EVENT_HPP
+#define MIGRAPHX_GUARD_MIGRAPHLIB_EVENT_HPP
+
+#include <migraphx/instruction.hpp>
+#include <migraphx/operators.hpp>
+#include <migraphx/config.hpp>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+struct create_events
+{
+    int num_of_events = 0;
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.num_of_events, "event"));
+    }
+    std::string name() const { return "gpu::create_events"; }
+    shape compute_shape(const std::vector<shape>& inputs) const
+    {
+        if(inputs.empty())
+            return {};
+        else
+            return inputs.front();
+    }
+
+    argument compute(context& ctx, const shape&, const std::vector<argument>&) const
+    {
+        ctx.create_events(num_of_events);
+        return {};
+    }
+};
+
+struct record_event
+{
+    int event = -1;
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.event, "event"));
+    }
+    std::string name() const { return "gpu::record_event"; }
+    shape compute_shape(const std::vector<shape>& inputs) const
+    {
+        if(inputs.empty())
+            return {};
+        else
+            return inputs.front();
+    }
+
+    argument compute(context& ctx, const shape&, const std::vector<argument>&) const
+    {
+        ctx.record_event(event);
+        return {};
+    }
+};
+
+struct wait_event
+{
+    int event = -1;
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.event, "event"));
+    }
+    std::string name() const { return "gpu::wait_event"; }
+    shape compute_shape(const std::vector<shape>& inputs) const
+    {
+        if(inputs.empty())
+            return {};
+        else
+            return inputs.front();
```

Comment:
This should always return empty. Why would it use the shape from the first parameter?

Reply:
In fact, this doesn't match what happens in the compute method where it always returns an empty object.